### PR TITLE
Use cita database executor

### PR DESCRIPTION
**Pull request by kaikai1024**
### Description of the Change

Rebase the branch: remove-snapshot. please review https://github.com/cryptape/cita/pull/700 first

TODO: *evm(delete?) and core*

### CODE REVIEW

File: cita-executor/src/main.rs [hunk 1]
```diff
@@ -87,7 +87,7 @@ extern crate common_types as types;
 extern crate core_executor as core;
 #[macro_use]
 extern crate crossbeam_channel;
-extern crate db as cita_db;
+extern crate cita_database as cita_db;
```
Comment (ouwenkg) [inline cita-executor/src/main.rs]: remove this `extern`?

File: cita-executor/src/backlogs.rs [hunk 2]
```diff
@@ -388,7 +387,7 @@ impl Backlogs {
     }
 
     pub fn completed_keys(&self) -> ::std::vec::Vec<&u64> {
-        self.completed.keys().sorted()
+        unimplemented!()
```
Comment (ouwenkg) [inline cita-executor/src/backlogs.rs]: Is this implementation related to db replacement?

File: cita-executor/src/main.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-executor/src/main.rs]: How to use `as` without `extern`?

File: cita-executor/src/backlogs.rs (hunk 2 above)
Comment (kaikai1024) [reply cita-executor/src/backlogs.rs]: I'll have another try.

File: cita-executor/core/src/trie_db.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            let value = values[i].clone();
+            cache.insert(key, value);
+        }
+        Ok(())
+    }
+
+    fn remove_batch(&self, _keys: &[Vec<u8>]) -> Result<(), Self::Error> {
+        Ok(())
+    }
+
```
Comment (ouwenkg) [inline cita-executor/core/src/trie_db.rs]: Not implement yet? Together with `remove()`

Comment (kaikai1024) [reply cita-executor/core/src/trie_db.rs]: ...
I'll use `unimplemented`.

File: cita-executor/src/main.rs (hunk 1 above)
Comment (ouwenkg) [reply cita-executor/src/main.rs]: Try [renaming-dependencies-in-cargotoml](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml)

Comment (leeyr338) [reply cita-executor/src/main.rs]: Try :
```rust
use cita_database as cita_db
```
?

File: cita-executor/core/src/trie_db.rs (hunk 3 above)
Comment (leeyr338) [reply cita-executor/core/src/trie_db.rs]: Yes, `unimplemented` is much better.

File: cita-executor/src/backlogs.rs (hunk 2 above)
Comment (ouwenkg) [reply cita-executor/src/backlogs.rs]: I am ok.

File: cita-executor/src/main.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-executor/src/main.rs]: > Try [renaming-dependencies-in-cargotoml](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#renaming-dependencies-in-cargotoml)

This works.

File: cita-executor/core/src/trie_db.rs (hunk 3 above)
Comment (kaikai1024) [reply cita-executor/core/src/trie_db.rs]: DONE.
It's the cods from old project...
